tbx/static_src/javascript/components/youtube-embed.js

```diff
@@ -0,0 +1,79 @@
+import Cookies from 'js-cookie';
+
+/**
+ * YouTubeConsentManager handles the consent for loading YouTube videos on a webpage.
+ */
+class YouTubeConsentManager {
+    static selector() {
+        return '.grid__embed.streamfield__embed.youtube-video-container';
+    }
+
+    /**
+     * Create a new YouTubeConsentManager.
+     */
+    constructor(node) {
+        this.consentButtonClass = 'consent-button';
+        this.dontAskAgainButtonClass = 'dont-ask-again-button';
+        this.videoPlaceholderClass = 'youtube-video-placeholder';
+        this.youtubeEmbedClass = 'youtube-embed';
+
+        this.youtubeEmbedContainer = node;
+        this.consentButton = this.youtubeEmbedContainer.querySelector(
+            `.${this.consentButtonClass}`,
+        );
+        this.dontAskAgainButton = this.youtubeEmbedContainer.querySelector(
+            `.${this.dontAskAgainButtonClass}`,
+        );
+        this.videoPlaceholder = this.youtubeEmbedContainer.querySelector(
+            `.${this.videoPlaceholderClass}`,
+        );
+        this.youtubeEmbed = this.youtubeEmbedContainer.querySelector(
+            `.${this.youtubeEmbedClass}`,
+        );
+
+        // Bind event handlers
+        this.consentButton.addEventListener(
+            'click',
+            this.handleconsentClick.bind(this),
+        );
+        this.dontAskAgainButton.addEventListener(
+            'click',
+            this.handleDontAskAgainClick.bind(this),
+        );
+
+        // Check if consent has been given previously
+        this.checkConsent();
+    }
+
+    loadYouTubeEmbed() {
+        // Hide the video placeholder
+        this.videoPlaceholder.style.display = 'none';
```

Comment:
we'd not usually update styling directly from the js, but instead update the classes and then change the CSS in a stylesheet. Again, I can tidy this one up.

Reply:
sorting in front-end ticket.